ocaml/typing/ctype.ml

```diff
@@ -5494,17 +5468,7 @@ let rec build_subtype env (visited : transient_expr list)
       let (t1', c1) = build_subtype env visited loops (not posi) level t1 in
       let (t2', c2) = build_subtype env visited loops posi level t2 in
       let (a', c3) =
-        if level > 2 then begin
-          (* If posi, then t1' >= t1, and we pick t1; otherwise we pick t1'. In
-            either case we pick the smaller type which is the "real" type of
-            runtime values, and easier to cross modes (and thus making the
-            mode-crossing more complete). *)
-          let t1 = if posi then t1 else t1' in
-          if is_immediate env t1 then
-            Mode.Alloc.newvar (), Changed
-          else
-            build_submode (not posi) a
-          end else a, Unchanged
```

Comment:
I think it might not be testable, because this is used by `enlarge_type`, which is used in `typecore.ml` but falls back to `subtype` which still crosses mode.

Reply:
I deleted it because it called `is_immediate`.... but the property of being "immediate" doesn't really exist any more, and I didn't immediately see how to translate this into the new world of modal kinds. So I deleted it to see what broke so that I might understand it better. Nothing broke! :)

Are you saying that this is an optimization? Is that why it should be added back?

Reply:
Yes I think it's an optimization - in `typecore.ml`, `enlarge_type` is called first, and if failed, `subtype` is called. There is still mode crossing in `subtype`, so the lack of mode crossing in `enlarge_type` is not testable. 

I think it's doable but I don't know how to keep it tested in the long term.

For now, maybe just leave a CR here?

Reply:
This isn't correct. `enlarge_type` is used when we are doing `(e :> t)` and we don't have (principally) know the type of `e`. Whereas `subtype` is used when we know the type of `e` (or if the user wrote `(e : t1 :> t2)`). I suspect something like this might be broken by this change:
```ocaml
module M : sig
  val foo : (local int -> unit) -> (int -> unit)
end = struct
  let foo f = (f :> int -> unit)
end
```

Reply:
thanks - I will add the missing logic with the test.

Reply:
Yes looks good. Thanks!